restcomm/restcomm.mgcp/src/main/java/org/restcomm/connect/mgcp/MockMediaGateway.java

```diff
@@ -89,30 +94,18 @@
     private RevolvingCounter connectionIdPool;
     private RevolvingCounter endpointIdPool;
 
-    private static Map<MediaSession, ActorRef> endpoints;
-    private static Map<MediaSession, ActorRef> links;
-    private static Map<MediaSession, ActorRef> connections;
+    private Map<ActorRef, ActorRef> connectionToEndpointMap;
+    private Map<String, String> connEndpointMap;
+
+    private ActorRef monitoringService;
 
     private ActorSystem system;
 
     public MockMediaGateway() {
         super();
-        endpoints = new ConcurrentHashMap<MediaSession, ActorRef>();
-        links = new ConcurrentHashMap<MediaSession, ActorRef>();
-        connections = new ConcurrentHashMap<MediaSession, ActorRef>();
         system = context().system();
-    }
-
-    public static Map<MediaSession, ActorRef> getEndpointsMap() {
-        return endpoints;
-    }
-
-    public static Map<MediaSession, ActorRef> getConnections() {
-        return connections;
-    }
-
-    public static Map<MediaSession, ActorRef> getLinks() {
-        return links;
+        connectionToEndpointMap = new ConcurrentHashMap<ActorRef, ActorRef>();
+        connEndpointMap = new ConcurrentHashMap<String, String>();
```

Comment:
considering this is Actor state, and concurrency should be already solved by Actor model, is there any particular intention on using ConcurrentHashMap??

Reply:
@jaimecasero no particular reason, just a habit